modules/process-testing/pages/contract.adoc

```diff
@@ -1,3 +1,146 @@
+
+== Retrieve contracts from files in the classpath
+
+Contracts can be stored in JSon files, and then retrieve at runtime from the classpath resources. This allows to extract the contract definition and the values in a dedicated file, which is easier to read and to maintain, especially for large and complex contracts. 
+
+A contract JSon file works as a key-value model, the key is the name of the contract input and the value the value of the contract input:
+
+[source, JSon]
+----
+{
+    "textContractInput" : "value",
+    "booleanContractInput" : true,
+    "dateContractInput" : "2021-10-10T00:00:00Z", <1>
+	"complexContractInput" : {
+		"integerContractInput" : 1,
+	 	"localDateTimeContractInput" : "2021-02-10T12:23:32",
+	},
+	"multipleFileContractInput" : [
+        "/documents/docA.txt",
+		"/documents/docB.txt"
+    ]
+}
+----
+<1> All dates (`LocalDate`, `LocaDateTime`, `OffsetDateTime` and `Date`) are passed as `String` using the https://en.wikipedia.org/wiki/ISO_8601[ISO_8601] standard format.
+
+Contract JSon files have to be available from the classpath at runtime. + 
+To do so, put them for example in a folder `src/test/resources/contracts/`, and then use them in your tests: 
+
+[source, Java]
+----
+var contract = ContractBuilder.newContract().fromClasspathResource("/contracts/<CONTRACT FILE NAME>.json");
+----
+
+== Test contract constraints
+
+A contract is made of contract inputs and constraints. Constraints are used to ensure the integrity of contracts inputs. +
+It is possible to validate in an integration test that for a given set of input a constraint fails:
+
+In the following example, we have a contract for a business object `Person`, with two contract inputs: a name and a birthdate. We also have a constraint on the birthdate: it cannot be in the futur. +
+The test validates that if the _birthdate_ contract input is a date in the futur, then the instanciation of the process fails.
+
+[source, Java]
+----
+@BonitaTests
+class ProcessIT {
+
+    @Test
+    void should_fail_on_birthdate_constraint(BonitaTestToolkit toolkit) {
+        User walter = toolkit.getUser("walter.bates");
+        ProcessDefinition processDef = toolkit.getProcessDefinition("myProcess");
+
+        var invalidContract = ContractBuilder.newContract()
+            .complexInput("personContractInput", ComplexInputBuilder.complexInput()
+                .textInput("name", "Adrien"))
+                .localDateInput("birthdate", LocalDate.now().plusDays(1))
+            .build();
+
+        assertThatThrownBy(() -> processDef.startProcessFor(walter, invalidContract))
+                        .isInstanceOf(StartProcessException.class)
+                        .hasMessageContaining("The input [birthdate] cannot be in the futur!");
```

Comment:
add a call-out explaining that the message is defined in the technical message field of the constraint